carta/image.py

```diff
@@ -672,6 +672,225 @@ def hide_contours(self):
+        if angular_source is None:
+            angular_source = self.macro("vectorOverlayConfig", "angularSource")
+        if intensity_source is None:
+            intensity_source = self.macro("vectorOverlayConfig", "intensitySource")
+        if pixel_averaging_enabled is None:
+            pixel_averaging_enabled = self.macro("vectorOverlayConfig", "pixelAveragingEnabled")
+        if pixel_averaging is None:
+            pixel_averaging = self.macro("vectorOverlayConfig", "pixelAveraging")
+        if fractional_intensity is None:
+            fractional_intensity = self.macro("vectorOverlayConfig", "fractionalIntensity")
+        if threshold_enabled is None:
+            threshold_enabled = self.macro("vectorOverlayConfig", "thresholdEnabled")
+        if threshold is None:
+            threshold = self.macro("vectorOverlayConfig", "threshold")
+        if debiasing is None:
+            debiasing = self.macro("vectorOverlayConfig", "debiasing")
+        if q_error is None:
+            q_error = self.macro("vectorOverlayConfig", "qError")
+        if u_error is None:
+            u_error = self.macro("vectorOverlayConfig", "uError")
+        self.call_action("vectorOverlayConfig.setVectorOverlayConfiguration", angular_source, intensity_source, pixel_averaging_enabled, pixel_averaging, fractional_intensity, threshold_enabled, threshold, debiasing, q_error, u_error)
+
+    @validate(NoneOr(Number()), NoneOr(Number(), Constant(Auto)), NoneOr(Number(), Constant(Auto)), NoneOr(Number()), NoneOr(Number()), NoneOr(Number()))
+    def set_vector_overlay_style(self, thickness=None, intensity_min=None, intensity_max=None, length_min=None, length_max=None, rotation_offset=None):
+        """Set the styling (line thickness, intensity range, line length range, rotation offset) of vector overlay.
+
+        Parameters
+        ----------
+        thickness : {0}
+            The line thickness in pixels. By default is 1.
+        intensity_min : {1}
+            The minimum value of intensity in Jy/pixel. Use :obj:`carta.constants.Auto.AUTO` to clear the custom value and calculate it automatically.
+        intensity_max : {2}
+            The maximum value of intensity in Jy/pixel. Use :obj:`carta.constants.Auto.AUTO` to clear the custom value and calculate it automatically.
+        length_min : {3}
+            The minimum value of line length in pixels. By default is 0.
+        length_max : {4}
+            The maximum value of line length in pixels. By default is 20.
+        rotation_offset : {5}
+            The rotation offset in degrees. By default is 0.
+        """
+        if thickness is not None:
+            self.call_action("vectorOverlayConfig.setThickness", thickness)
+        if intensity_min is None:
+            intensity_min = self.macro("vectorOverlayConfig", "intensityMin")
+        elif intensity_min is Auto.AUTO:
+            intensity_min = Undefined()
+        else:
+            intensity_min = intensity_min
```

Comment:
This is unnecessary -- you're setting the variable to its own value.

Reply:
Done in 142c3b7.

Reply:
@I-Chenn what's the idea of having this kind of statement? Please elaborate.

Reply:
With the intention of listing all the possible values to clarify the logic of the code for developers/users, though I know it's stupid of setting a variable to its own value. Sorry for that